### PR TITLE
Added integration docs for CircleCI

### PR DESCRIPTION
**Pull request by michaelnji**
Added integration docs for CircleCI using the official node orb for circleci

### CODE REVIEW

File: docs/integration/circle.md [hunk 1]
```diff
@@ -1,3 +1,51 @@
 # Integrating with CircleCI
 
-This is a stub article. Help writing it by clicking "Edit this page on GitHub"
+[CricleCI](https://circleci.com/) is a continous integration platform that allows you test your code and automate workflows in your project. You can use StepCI in CircleCI using the official [Node Orb](https://circleci.com/developer/orbs/orb/circleci/node)
+
+To integrate StepCI into your CircleCI build:
```
Comment (mishushakov) [inline docs/integration/circle.md:5]: Write Step CI separately

**Review by mishushakov — CHANGES_REQUESTED** [review]
Looks good, only spelling needs to be corrected. Have you also tried running it? Did it work?

File: docs/integration/circle.md [hunk 2]
```diff
@@ -1,3 +1,51 @@
...
+jobs:
+  install-and-run:
+    executor: node/default 
+    steps:
+      - checkout
+      - node/install-packages:
+          pkg-manager: npm
+      - run:
+          command: npm install -D stepci # install stepci as a dev dependency
+          name: install StepCI
```
[the remark this answers is not in the document]
Comment (michaelnji) [reply docs/integration/circle.md:40]: It seems not to work as a global dependency as I get a `command not found: stepci` error. Installing as a dev dependency resolves this
```

File: docs/integration/circle.md (hunk 1 above)
Comment (michaelnji) [reply docs/integration/circle.md:5]: Okay let me change that

Comment (michaelnji) [reply docs/integration/circle.md:5]: It runs well in the nodejs orb on CircleCI. I've also tested with the bun orb (with a different config) and that works too.

**Review by michaelnji — COMMENTED** [review]
All typos fixed

**Comment by mishushakov** [comment]
Looks good to me. Thank you very much!

**Pull request merged**